File: src/main/java/edu/harvard/mcz/imagecapture/interfaces/ScanCounterInterface.java
package edu.harvard.mcz.imagecapture.interfaces;

import edu.harvard.mcz.imagecapture.jobs.RunnableJobError;

import java.util.List;

public interface ScanCounterInterface {
    void logBarcode(String barcode);

    void appendError(RunnableJobError anError);

    String getErrorReport();

    void incrementSpecimenDatabased();

    void incrementSpecimenExisting();

    void incrementTotal();

    void incrementFilesSeen();

    void incrementFilesFailed();

    void incrementFilesDatabased();

    void incrementFilesExisting();

    void incrementDirectories();

    void incrementDirectoriesFailed();

    int getSpecimens();

    int getTotal();

    /**
     * @return the files checked
     */
    int getFilesSeen();

    /**
     * @return the files for which new database records were created
     */
    int getFilesDatabased();

    /**
     * @return the files for which database records allready existed
     */
    int getFilesExisting();

    /**
     * @return the files for which a check for information needed to create a record Failed.
     */
    int getFilesFailed();


    /**
     * @return the directories
     */
    int getDirectories();


    /**
     * @return the directoriesFailed
     */
    int getDirectoriesFailed();

    /**
     * @return the specimensUpdated
     */
    int getSpecimensUpdated();

    int getSpecimensExisting();

    /**
     *
     */
    void incrementSpecimensUpdated();

    /**
     * @return the filesUpdated
     */
    int getFilesUpdated();

    /**
     *
     */
    void incrementFilesUpdated();

    /**
     * @return
     */
    List<RunnableJobError> getErrors();

    List<String> getBarcodes();

}

File: src/main/kotlin/edu/harvard/mcz/imagecapture/interfaces/ScanCounterInterface.kt
package edu.harvard.mcz.imagecapture.interfaces


import edu.harvard.mcz.imagecapture.jobs.RunnableJobError

interface ScanCounterInterface {
    fun appendError(anError: RunnableJobError?)
    val errorReport: String
    fun incrementSpecimenDatabased()
    fun incrementSpecimenExisting()
    fun incrementTotal()
    fun incrementFilesSeen()
    fun incrementFilesFailed()
    fun incrementFilesDatabased()
    fun incrementFilesExisting()
    fun incrementDirectories()
    fun incrementDirectoriesFailed()
    val specimens: Int
    val total: Int
    /**
     * @return the files checked
     */
    val filesSeen: Int

    /**
     * @return the files for which new database records were created
     */
    val filesDatabased: Int

    /**
     * @return the files for which database records allready existed
     */
    val filesExisting: Int

    /**
     * @return the files for which a check for information needed to create a record Failed.
     */
    val filesFailed: Int

    /**
     * @return the directories
     */
    val directories: Int

    /**
     * @return the directoriesFailed
     */
    val directoriesFailed: Int

    /**
     * @return the specimensUpdated
     */
    val specimensUpdated: Int

    val specimensExisting: Int
    /**
     *
     */
    fun incrementSpecimensUpdated()

    /**
     * @return the filesUpdated
     */
    val filesUpdated: Int

    /**
     *
     */
    fun incrementFilesUpdated()

    /**
     * @return
     */
    val errors: MutableList<Any?>?
}
